tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -290,6 +408,131 @@ static void test_rx_suspend_disable_resume(void)
 	PASS();
 }
 
+static void test_tx_gatt_suspend_resume(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+	ASSERT_OK_MSG(bt_mesh_prov_enable(BT_MESH_PROV_GATT), "Failed to enable GATT provisioner");
+	/* Let the tester observe pb gatt advertisements before provisioning. The node should
+	 * advertise pb gatt service with 100 msec (ext adv) or 1000msec (legacy adv) interval.
+	 */
+	k_sleep(K_MSEC(1800));
+
+	LOG_INF("Provisioning device under test");
+	bt_mesh_provision(test_net_key, 0, 0, 0, tx_cfg.addr, tx_cfg.dev_key);
+
+	/* Let the tester observe proxy advertisements */
+	k_sleep(K_MSEC(6500));
+
+	/* Send a mesh message to notify tester that device is suspended */
+	struct bt_mesh_adv *adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+			BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, "1", sizeof("1"));
+	bt_mesh_adv_send(adv, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	ASSERT_OK(bt_mesh_suspend());
+	LOG_INF("Mesh suspended.");
+
+	k_sleep(K_SECONDS(SUSPEND_DURATION));
+
+	ASSERT_OK(bt_mesh_resume());
+	LOG_INF("Mesh resumed.");
+
+	/* Send a mesh message to notify tester that device is resumed */
+	adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+			BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, "0", sizeof("0"));
+	bt_mesh_adv_send(adv, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	/* Let the tester observe that proxy advertisement resumes */
+	k_sleep(K_MSEC(6000));
+
+	PASS();
+}
+
+static void test_tx_gatt_suspend_disable_resume(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+	ASSERT_OK_MSG(bt_mesh_prov_enable(BT_MESH_PROV_GATT), "Failed to enable GATT provisioner");
+	/* Let the tester observe pb gatt advertisements before provisioning. The node should
+	 * advertise pb gatt service with 100 msec (ext adv) or 1000msec (legacy adv) interval.
+	 */
+	k_sleep(K_MSEC(1800));
+
+	LOG_INF("Provisioning device under test");
+	bt_mesh_provision(test_net_key, 0, 0, 0, tx_cfg.addr, tx_cfg.dev_key);
+
+	/* Let the tester observe proxy advertisements */
+	k_sleep(K_MSEC(6500));
+
+	/* Send a mesh message to notify tester that device is suspended */
+	struct bt_mesh_adv *adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+			BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, "1", sizeof("1"));
+	bt_mesh_adv_send(adv, NULL, NULL);
+	k_sleep(K_MSEC(150));
```

Comment:
Wrap into a function?

Reply:
This is purely used as a sync mechanism right? From what I can see we do not have anything like this in the common modules for bsim tests. IMO if we are adding this sort of sync mechanism, it should be added to the common library so we don't end up re-implementing it for other tests. I have considered something like this before myself.

Reply:
Yes, only used as a sync mechanism. 
I will look into adding it to the common lib. Should we have a fixed message size (i. e. 8/16-bit) and retransmit count, or let these be specified by the user?

Reply:
I added two functions to `mesh_test.c/h` for the message passing/sync. May be hard to look at the diff since I also did a rebase, but the changes are in `mesh_test`, `gatt_common` and the test scripts where I use the updated functions.